Realm/RLMUtil.mm

```diff
@@ -421,9 +417,19 @@ BOOL RLMIsRunningInPlayground() {
         case RLMPropertyTypeObject: {
             // If we are unboxing an object and it is unmanaged, we need to
             // add it to the Realm.
-            if (RLMObjectBase *objBase = RLMDynamicCast<RLMObjectBase>(v); !objBase->_realm) {
-                RLMAddObjectToRealm(objBase, realm, RLMUpdatePolicyError);
+            if (RLMObjectBase *objBase = RLMDynamicCast<RLMObjectBase>(v); !objBase->_realm
+                && createPolicy.create) {
+                RLMVerifyInWriteTransaction(realm);
+                // We do not want to copy the object, instead we want to promote
+                // it to a managed one.
+                createPolicy.copy = false;
```

Comment:
Shouldn't this just forward copy on as-is? We do want to copy the object if we're inside a call to `create()`.

Reply:
true, I changed this to actually honour the create policy